File: Recursion/Python/bubbleSort.py

arr = [20,50,30,22,99,66,44]

def bubbleSort(arr,n):
    if n == 1:
        return

    for i in range(n):
        if arr[i] > arr[i+1]:
            arr[i],arr[i+1] = arr[i+1],arr[i]
    bubbleSort(arr,n-1)


bubbleSort(arr,len(arr)-1)

print(arr)

File: Recursion/JavaScript/bubbleSort.js
arr = [34,23,12,88,55,43]

function bubbleSort(arr,n){
    if(n===1){
        return
    }
    for(let i=0;i<=n-1;i++){
        if(arr[i]>arr[i+1]){
            const temp = arr[i]
            arr[i] = arr[i+1]
            arr[i+1] = temp
        }
    }
    bubbleSort(arr,n-1)
}

bubbleSort(arr,arr.length - 1)

console.log(arr);